### ISSUE TITLE
Swift Audit

### ISSUE THREAD

**Issue by apbendi** [cloudmine/CMHealthSDK-iOS#11]
Spike out a project in Swift that uses the API and ensure that it feels "Swifty enough". Look for any sharp edges that could be smoothed easily.

**Comment by Splat** [comment]
'Must call a designated initializer of the superclass’ in Swift when subclassing CMObject. It’s not an issue in 1.7.8, but is in 1.7.10. I believe the answer to be convenience methods on the init methods which set variables not present in the params. To be clear:

```
(instancetype)initWithObjectId:(NSString *)theObjectId NS_DESIGNATED_INITIALIZER;
convenience (instancetype)initWithObjectId:(NSString *)theObjectId NS_DESIGNATED_INITIALIZER;
```
and
```
(instancetype)initWithCoder:(NSCoder *)aDecoder;
convenience (instancetype)initWithCoder:(NSCoder *)aDecoder;
```

I don't believe the default init override needs to be a convenience method and can be the default that each of the convenience initializers call at some point. So as long as ```initWithObjectId``` and ```initWithCoder``` both are convenience methods and both leverage ```init``` at some point. The ultimate reason I am not trying to implement this is currently ```init``` and ```initWithCoder``` both leverage ```initWithObjectId``` and from my reading on Swift constructors, ```initWithObjectId``` should be a convenience. Without having a truly modern Swift/iOS context it's better in the hands of a master. Basically our constructors need some shuffling and some testing. Ideally everything should be backwards compat so updating the SDK doesn't become an order of operations situation.

**Comment by apbendi** [comment]
hey @Splat, thanks for the info. Did this come from a customer or did you encounter it with an internal project? Also, looks like this applies to the generic CM SDK, not to CMHealth. Mind moving it to a new issue there and we can discuss further? Thanks!

**Comment by apbendi** [comment]
This issue on the main SDK was resolved with the 1.7.11 release.
